Scenarios/CGCT/CGCT/PathDemoTableViewCell.m

```diff
@@ -0,0 +1,189 @@
+    CGPathAddLineToPoint(ribbon, NULL, bounds.size.width * .1, bounds.size.height * .86);
+    CGPathAddArc(ribbon, NULL, bounds.size.width * .06, bounds.size.height * .81, bounds.size.width * .09, M_PI * .4, M_PI * 1.9, true);
+    CGPathAddArc(ribbon, NULL, bounds.size.width * .21, bounds.size.height * .75, bounds.size.width * .07, M_PI * .9, M_PI * 1.32, false);
+    CGPathAddLineToPoint(ribbon, NULL, bounds.size.width * .23, bounds.size.height * .65);
+    CGPathAddLineToPoint(ribbon, NULL, bounds.size.width * .2, bounds.size.height * .64);
+    CGPathCloseSubpath(ribbon);
+    CGContextAddPath(context, ribbon);
+
+    CGAffineTransform reverseRibbon = CGAffineTransformIdentity;
+    reverseRibbon = CGAffineTransformScale(reverseRibbon, -1, 1);
+    reverseRibbon = CGAffineTransformTranslate(reverseRibbon, -bounds.size.width, 0);
+    CGPathRef reversedRibbon = CGPathCreateCopyByTransformingPath(ribbon, &reverseRibbon);
+    CGContextAddPath(context, reversedRibbon);
+
+    CGContextClip(context);
+
+    CGRect gradientRect = CGRectMake(bounds.size.width * .05, bounds.size.height * .6, bounds.size.width * .8, bounds.size.height * .4);
+    CGPoint startPoint = CGPointMake(CGRectGetMinX(gradientRect), CGRectGetMaxY(gradientRect));
+    CGPoint endPoint = CGPointMake(CGRectGetMaxX(gradientRect), CGRectGetMinY(gradientRect));
+
+    CGContextDrawLinearGradient(context, gradient, startPoint, endPoint, 0);
+    CGGradientRelease(gradient), gradient = NULL;
+
+    CGContextRestoreGState(context);
+
+    CGContextAddPath(context, ribbon);
+    CGContextAddPath(context, reversedRibbon);
+    CGContextDrawPath(context, kCGPathStroke);
+    CGPathRelease(ribbon);
+    CGPathRelease(reversedRibbon);
+
+    CGMutablePathRef textPath = CGPathCreateMutable();
+    CGPathAddRect(textPath, NULL, CGRectMake(0, 0, bounds.size.width, bounds.size.height * .3));
+
+    CGContextSaveGState(context);
+    CGContextScaleCTM(context, 1.0f, -1.0f);
+    CGContextTranslateCTM(context, bounds.size.width * .25, -bounds.size.height);
+
+    CTFontRef myCFFont = CTFontCreateWithName((__bridge CFStringRef) @"Segoe UI", bounds.size.height * .1, NULL);
+    CFAutorelease(myCFFont);
+
+    NSDictionary* attributesDict = [NSDictionary dictionaryWithObjectsAndKeys:(__bridge id)myCFFont,
+                                                                              (id)kCTFontAttributeName,
+                                                                              [UIColor blackColor].CGColor,
+                                                                              (id)kCTForegroundColorAttributeName,
+                                                                              nil];
+
+    CFAttributedStringRef attrString =
+        CFAttributedStringCreate(kCFAllocatorDefault, (__bridge CFStringRef) @"Core Demo", (__bridge CFDictionaryRef)attributesDict);
```

Comment:
I don't think we need to cram CT use into a mostly CG test right now; we can definitely have more different kinds of tests. The point of this app is to showcase the cool things we can do, yes, but not necessarily all at the same time :)